services/src/main/java/org/exoplatform/addons/gamification/service/configuration/RealizationsServiceImpl.java

```diff
@@ -36,7 +39,10 @@ public List<GamificationActionsHistoryDTO> getAllRealizationsByFilter(Realizatio
     if (fromDate.after(toDate)) {
       throw new IllegalArgumentException("Dates parameters are not set correctly");
     }
-    return realizationsStorage.getAllRealizationsByFilter(filter, offset, limit);
+    if (earnerId <= 0 && !isAdministrator(identity)) {
+      return realizationsStorage.getAllRealizationsByFilter(filter, offset, limit);
+    }
+    return realizationsStorage.getUsersRealizationsByFilter(filter, offset, limit);
```

Comment:
```suggestion
    if (filter.getEarnerId() > 0 && !isAdministrator(identity)) {
      throw new IllegalAccessException("User doesn't have enough privileges to access all achievements");
    }
    return realizationsStorage.getUsersRealizationsByFilter(filter, offset, limit);
```

Reply:
change made, and added another condition for the case of administrator get**All**RealizationsByFilter